pkg/util/azureclient/mgmt/features/deployments.go

```diff
@@ -14,6 +14,7 @@ import (
 // DeploymentsClient is a minimal interface for azure DeploymentsClient
 type DeploymentsClient interface {
 	Get(ctx context.Context, resourceGroupName, deploymentName string) (mgmtfeatures.DeploymentExtended, error)
+	Delete(ctx context.Context, resourceGroupName string, deploymentName string) (result mgmtfeatures.DeploymentsDeleteFuture, err error)
```

Comment:
For anything that returns a Future, write a DeleteAndWait function instead (grep to see what I mean) and put that in the Addons interface instead.  Rationale: simplifies unit testing because you just have to unit test one call, not two.